SERVICE: Authentification -> Authorize feature

linakaci opened the pull request.

linakaci committed:
service(add): authentification domain with the according structures and functions

linakaci committed:
database(init): init mongodb as a distant db with the connection and collection get functions

linakaci committed:
add(gitignore): with the .env and the binary

linakaci committed:
change(arch): add interfaces and structs to structure the API with database, usecase handling

linakaci committed:
add config struct for magic values (temporary)

linakaci committed:
fix(norme): generic func database, remove timout from context, call generic func to main

linakaci committed:
add(auth): authorize function that requests the user token and returns the userID
http handler authorise function
usecase method authorize

linakaci committed:
add(func): findByID domain function

linakaci committed:
update(Errmsg): update error message in some functions

linakaci committed:
update(Errmsg): some error message

Encorpluptit commented on authentification/domain/authentification.go:
```diff
@@ -38,7 +37,7 @@ type Auth struct {
 	bytes, err := bcrypt.GenerateFromPassword([]byte(password), 14)
 	if err != nil {
-		log.Fatal(err)
+		return "", fmt.Errorf("Password error: %v", err.Error())
```
pas besoin de .Error(), %v le fait tout seul

ou alors:
```suggestion
		return "", fmt.Errorf("Password error: %s", err.Error())
```

Encorpluptit commented on authentification/transport/http/authorize.go:
```diff
@@ -0,0 +1,28 @@
+
+		err := c.BodyParser(&token)
+		if err != nil {
+			c.Status(http.StatusBadRequest).JSON(err.Error())
```
laisse l'encodeur Json s'occuper de jsonify ton erreur.
ça change rien si c'est déjà une string, mais si c'est un structure, c'est beaucoup plus pratique pour personnes qui recoivent ton erreur.

On reverra très probablement les erreurs comme Anton l'a fait et te l'a suggéré dans une précédente PR.

Encorpluptit commented on authentification/usecase/login.go:
```diff
@@ -19,24 +18,20 @@ func (u UseCase) Login() LoginProto {
 		auth, err := u.DB.FindByEmail(ctx, input.Email)
 		if err != nil {
-			log.Println("Email or Password are incorrect")
 			return nil, err
```
Soit bien sûr que tes erreurs ont bien été wrap pour donner un context.

J'ai pas check du tout, donc si c'est le cas, good !

linakaci committed:
fix(ErrMsg): remove the .Error() from the error message

linakaci committed:
fix(ErrMsg): remove the .Error() func in the error message

linakaci committed:
Update authorize.go

linakaci committed:
add(features), change(ErrMsg):
Handler file like Anton's handler file with the Error and Success Response handler,
Call the Handler functions in the http handlers

linakaci committed:
rm(msg): remove unused log message

Encorpluptit reviewed (approved):
LGTM
